DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionSwizzler.swift

```diff
@@ -12,6 +12,8 @@ internal class URLSessionSwizzler {
     /// This value will increment for each call to the `bind()` method.
     /// Calling `unbind()` will decrement the count, when reaching zero, the swizzler is disabled.
     internal private(set) static var bindingsCount: Int = 0
+    /// The binding lock.
```

Comment:
non blocking

we should either make it `UInt` as we never want it to be negative by a developer error.

Or log something when`unbind` > `bind` calls. There has to something wrong in this case IMO.

Reply:
Good point, I will make it `UInt` 👍